java-frontend/src/test/java/org/sonar/java/resolve/GenericsTest.java

```diff
@@ -853,46 +854,15 @@ public void test_method_resolution_with_substitution_in_type_hierarchy() {
   }
 
   @Test
-  public void test_method_resolution_for_parametrized_method_with_type_variable_inheritance2() {
-    List<Type> elementTypes = declaredTypes(
-      "class PreTest<T> {"
-        + "  <S extends T> void foo(S s) {}"
-        + "}"
-
-        + "class Test<T> extends PreTest<T> {"
-        + "  void test() {"
-        + "    new Test<A>().<A>foo(new A());"
-        + "    new Test<A>().<B>foo(new B());"
-        + "    new Test<A>().foo(new A());"
-        + "    new Test<A>().foo(new B());"
-        + "  }"
-        + "}"
-
-        + "class Test2 extends Test<A> {"
-        + "  void test2() {"
-        + "    new Test2().<A>foo(new A());"
-        + "    new Test2().<B>foo(new B());"
-        + "    new Test2().foo(new A());"
-        + "    new Test2().foo(new B());"
-        + "  }"
-        + "}"
-
-        + "class Test3 extends Test2 {"
-        + "  void test3() {"
-        + "    new Test3().<A>foo(new A());"
-        + "    new Test3().<B>foo(new B());"
-        + "    new Test3().foo(new A());"
-        + "    new Test3().foo(new B());"
-        + "  }"
-        + "}"
-
-        + "class A {}"
-        + "class B extends A {}");
+  public void parametrized_method_resolution_with_bounded_type_variable() {
+    List<Type> elementTypes = declaredTypesFromFile("src/test/files/resolve/GenericMethodsBoundedTypeVariables.java");
 
     JavaType type = (JavaType) elementTypes.get(0);
     JavaSymbol.MethodJavaSymbol methodSymbol = getMethodSymbol(type, "foo");
-    // FIXME should be 12
-    assertThat(methodSymbol.usages()).hasSize(0);
+    // FIXME should be 12 - explicit type arguments not handled
```

Comment:
there is already a ticket about that missing piece. could be great to reference it.